### PR TITLE
Setup CI. Add documentation and example of usage

### LINKED ISSUE #920: Create `membrane_transcoder_plugin` package
We should create a new `membrane_transcoder_plugin` package that provides transcoding functionalities for audio and video streams.

TODO:
- [] Move transcoder's code from Boombox to a separate repository
- [ ] Add tests
- [ ] Add usage examples and documentation
- [ ] Release `membrane_transcoder_plugin` on Hex

### PR DESCRIPTION
**Pull request by varsill**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,18 @@ def deps do
 end
 ```
 
+## Usage
+In the `examples/vp8_to_h264.exs` file there is and example showing how to use
```
Comment (FelonEkonom) [inline README.md]:
```suggestion
In the `examples/vp8_to_h264.exs` file there is an example showing how to use 
```

File: examples/vp8_to_h264.exs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      |> child(:sink, %Membrane.File.Sink{location: output_file})
+
+    RCPipeline.subscribe(pipeline, _any)
+    RCPipeline.exec_actions(pipeline, spec: spec)
+    RCPipeline.await_end_of_stream(pipeline, :sink)
+    RCPipeline.terminate(pipeline)
+  end
+end
+File.mkdir("tmp")
+Example.convert(Path.join("./test/fixtures", "video.ivf"), Path.join("./tmp", "video.h264"))
```
Comment (FelonEkonom) [inline examples/vp8_to_h264.exs]: format this file

File: lib/transcoder.ex [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+  * `Membrane.H264`
+  * `Membrane.H265`
+  * `Membrane.VP8`
+  * `Membrane.RawVideo`
+
+  The following audio stream formats are supported:
+  * `Membrane.AAC`
+  * `Membrane.Opus`
+  * `Membrane.RawAudio`
+  * `Membrane.RemoteStream{content_type: Membrane.Opus}`
```
Comment (FelonEkonom) [inline lib/transcoder.ex]: This is supported only as the input

File: lib/transcoder.ex [hunk 4]
```diff
@@ -1,5 +1,23 @@
...
+
+  Please note that not all the conversions are possible!
```
Comment (FelonEkonom) [inline lib/transcoder.ex]: What conversions are not possible?

File: lib/transcoder.ex [hunk 5]
```diff
@@ -30,7 +57,15 @@ defmodule Membrane.Transcoder do
...
   def_options output_stream_format: [
-                spec: stream_format() | stream_format_module() | stream_format_resolver()
+                spec: stream_format() | stream_format_module() | stream_format_resolver(),
+                description: """
+                An option specifying desired output format.
+                Can be either:
+                * a struct being a Membrane stream format,
+                * a module in which Membrane stream format struct is defined,
+                * a function which receives input stream format as an input argument
+                and is supposed to return the desired output stream format.
```
Comment (FelonEkonom) [inline lib/transcoder.ex]:
```suggestion
                and is supposed to return the desired output stream format or its module.
```

**Review by FelonEkonom — CHANGES_REQUESTED** [review]

File: lib/transcoder.ex (hunk 4 above)
Comment (varsill) [reply lib/transcoder.ex]: H264 to AAC :p

File: examples/vp8_to_h264.exs [hunk 6]
```diff
@@ -5,9 +5,9 @@ Mix.install([
 ])
 
 defmodule Example do
-alias Membrane.RCPipeline
-require Membrane.RCPipeline, as: RCPipeline
-import Membrane.ChildrenSpec
+  alias Membrane.RCPipeline
+  require Membrane.RCPipeline, as: RCPipeline
```
Comment (FelonEkonom) [inline examples/vp8_to_h264.exs:9]: Let's add `examples/.*.exs` to the `.formatter.exs`

File: lib/transcoder.ex (hunk 4 above)
Comment (FelonEkonom) [reply lib/transcoder.ex]: So I would write that if the input stream format is audio, the output stream format must be audio as well (this same if the type is video). The way it is written right now suggests that e.g. we might not be able to transcode VP8 to H264

**Review by FelonEkonom — CHANGES_REQUESTED** [review]

File: lib/transcoder.ex (hunk 4 above)
Comment (varsill) [reply lib/transcoder.ex]: Well, perhaps I will just remove that note, the more I think about it the more I am inclining towards conclusion that it is obvious :D

**Review by FelonEkonom — APPROVED** [review]

**Pull request merged**